iree/compiler/InputConversion/Common/ConvertUpstreamToIREE.cpp

```diff
@@ -83,12 +83,39 @@ struct ConvertTensorCastPattern : public OpConversionPattern<tensor::CastOp> {
   }
 };
 
+struct ConvertTensorFromElementsPattern
+    : public OpConversionPattern<tensor::FromElementsOp> {
+  using OpConversionPattern<tensor::FromElementsOp>::OpConversionPattern;
+
+  LogicalResult matchAndRewrite(
+      tensor::FromElementsOp op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const override {
+    auto loc = op.getLoc();
+
+    if (shouldBeConverted(op)) {
+      SmallVector<Value> dimSizes(1);
+      dimSizes[0] = rewriter.create<ConstantIndexOp>(loc, 1);
+      rewriter.replaceOpWithNewOp<IREE::Flow::TensorSplatOp>(
+          op, op.getType(), operands.front(), dimSizes);
+    }
+
+    return success();
+  }
+
+  // TODO: This pattern was mainly added to iron out some kinks specific to
+  // detensoring (see: https://github.com/google/iree/issues/1159). Do we need
+  // to expand this check for other uses?
+  static bool shouldBeConverted(tensor::FromElementsOp op) {
```

Comment:
I think you need to check that it is static here too and that it is rank 1 (otherwise if it's `tensor<f32>` this will crash). It'd be nice if it supported the full cases but if this unblocks the work then it's ok to start. You may want to add some negative tests that ensure that ones with dynamic shapes, non-1 ranks, etc don't get converted instead of crashing though :)

Reply:
I think the result type of `tensor::FromElementsOp` is always a 1D tensor, right?
https://mlir.llvm.org/docs/Dialects/TensorOps/#tensorfrom_elements-mlirtensorfromelementsop.
I think we don't need these checks here.

Reply:
Eww, that's a silly op - even more reason to convert out of it ASAP. No clue why that would be limited to 1d, and it explains why they are always followed by a reshape. Yuck.